ui/src/features/project/list/projects-list.tsx

```diff
@@ -31,23 +65,30 @@ export const ProjectsList = () => {
     })
   });
 
+  const handlePaginationChange = (newPage: number, newPageSize: number) => {
+    setPage(newPage);
+    setPageSize(newPageSize);
+  };
+
+  const handleFilterChange = (newFilter: string) => {
+    setFilter(newFilter);
+    setPage(1);
+  };
+
   if (isLoading) return <LoadingState />;
 
   if (!data || data.projects.length === 0) return <Empty />;
 
   return (
     <>
       <div className='flex items-center mb-6'>
-        <ProjectListFilter onChange={setFilter} init={filter} />
+        <ProjectListFilter onChange={handleFilterChange} init={filter} />
         <Pagination
           total={data?.total || 0}
           className='ml-auto flex-shrink-0'
           pageSize={pageSize}
           current={page}
-          onChange={(page, pageSize) => {
-            setPage(page);
-            setPageSize(pageSize);
-          }}
+          onChange={handlePaginationChange}
```

Comment:
You would need to add `showSizeChanger` option to show page size filter

Reply:
please confirm if this is what you requested.

Reply:
Yes it is what I expected. Thanks